CI failed in `Compare to postgres`

**Describe the bug**
A clear and concise description of what the bug is.

**To Reproduce**
Steps to reproduce the behavior:

**Expected behavior**
A clear and concise description of what you expected to happen.

**Additional context**
Add any other context about the problem here.

Example failure:

```
=================================== FAILURES ===================================
_ TestPsqlParity.test_sql_file[/home/runner/work/arrow-datafusion/arrow-datafusion/integration-tests/sqls/values_list.sql] _

self = <test_psql_parity.TestPsqlParity object at 0x7ff3ad617d30>
fname = PosixPath('/home/runner/work/arrow-datafusion/arrow-datafusion/integration-tests/sqls/values_list.sql')

    @pytest.mark.parametrize("fname", test_files, ids=str)
    def test_sql_file(self, fname):
        datafusion_output = pd.read_csv(io.BytesIO(generate_csv_from_datafusion(fname)))
        psql_output = pd.read_csv(io.BytesIO(generate_csv_from_psql(fname)))
>       np.testing.assert_allclose(datafusion_output, psql_output, equal_nan=True, verbose=True)

integration-tests/test_psql_parity.py:91: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/hostedtoolcache/Python/3.10.8/x[64](https://github.com/apache/arrow-datafusion/actions/runs/3503739074/jobs/5869029028#step:9:65)/lib/python3.10/site-packages/numpy/__init__.py:309: in __getattr__
    import numpy.testing as testing
/opt/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/site-packages/numpy/testing/__init__.py:12: in <module>
    from ._private import extbuild, decorators as dec
/opt/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/site-packages/numpy/testing/_private/extbuild.py:11: in <module>
    from numpy.distutils.ccompiler import new_compiler
/opt/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/site-packages/numpy/distutils/__init__.py:24: in <module>
    from . import ccompiler
/opt/hostedtoolcache/Python/3.10.8/x64/lib/python3.10/site-packages/numpy/distutils/ccompiler.py:20: in <module>
    from numpy.distutils import log
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
```

Maybe some package was updated in the python ecosystem and we need to update the harness to keep up 🤔 


@Jimexist  any ideas?

--- a/.github/workflows/rust.yml
+++ b/.github/workflows/rust.yml
@@ -194,7 +194,14 @@ jobs:
           python-version: "3.10"
       - name: Install Python dependencies
         run: |
-          python -m pip install --upgrade pip setuptools wheel
+          python -m pip install --upgrade pip wheel
+          # setuptools is normally only intended to serve as a build-time dependency
+          # but since numpy uses it internally and currently accessing methods that
+          # are removed from newer versions of setuptools we have a pin on it as well.
+          #
+          # TODO: this should be removed as soon as https://github.com/numpy/numpy/issues/22623
+          # is resolved and numpy version in integration-tests/requirements.txt is updated.
+          python -m pip install --upgrade "setuptools<=65.5.1"
           python -m pip install -r integration-tests/requirements.txt
       - name: Allow access of psql
         run: |

